intel_extension_for_transformers/llm/runtime/graph/vectors/cpu/vec_base.hpp

```diff
@@ -54,16 +68,49 @@ typedef std::pair<__m256i, __m256i> int32x16;
   }
 
 #endif
-typedef __m256i bf16x16;
-typedef __m256i int16x16;
-typedef __m128i int8x16;
+
+struct bf16x16 {
+  __m256i first;
+};
+
+struct fp16x16 {
+  __m256i first;
+};
+
+struct s16x16 {
+  __m256i first;
+};
+struct s8x16 {
+  __m128i first;
+};
+struct u8x16 {
+  __m128i first;
+};
+
 #define CPU_VEC_STEP 16
 
 template <typename T>
-T load_kernel_t(const void*);
+T load_kernel_t(const void* src) {
+  return *reinterpret_cast<const T*>(src);
+}
+
+template <>
+fp32x16 load_kernel_t<fp32x16>(const void* src);
+template <>
+bf16x16 load_kernel_t<bf16x16>(const void* src);
 
 template <typename T>
-void store_kernel_t(void*, T);
+void store_kernel_t(void* dst, T src) {
+  T* dst_T = reinterpret_cast<T*>(dst);
+  *dst_T = src;
+}
+
+template <>
+void store_kernel_t<s8x16>(void* dst, s8x16 src);
+template <>
+void store_kernel_t<fp32x16>(void* dst, fp32x16 src);
+template <>
+void store_kernel_t<bf16x16>(void* dst, bf16x16 src);
```

Comment:
There seems to be duplicated definitions here.
![image](https://github.com/intel/intel-extension-for-transformers/assets/28386673/98925a09-60a0-442b-b96c-b7a85c291cd6)

Reply:
No duplication here. Line 103 is the declarations and definitions for the template function, and it has some specializations for some type. The declaration of specializations are in vec_base.hpp while their definitions are in vec_store.hpp.

Reply:
You are right, but MSVC tolds me that there are duplications about store_kernel_t.
``` bat
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_convert.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct s8x16>(void *,struct s8x16)" (??$store_kernel_t@Us8x16@@@@YAXPEAXUs8x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_convert.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct fp32x16>(void *,struct fp32x16)" (??$store_kernel_t@Ufp32x16@@@@YAXPEAXUfp32x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_convert.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct bf16x16>(void *,struct bf16x16)" (??$store_kernel_t@Ubf16x16@@@@YAXPEAXUbf16x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_arithmetic.cpp.obj : warning LNK4006: "struct fp32x16 __cdecl load_kernel_t<struct fp32x16>(void const *)" (??$load_kernel_t@Ufp32x16@@@@YA?AUfp32x16@@PEBX@Z) already defined in vec_load.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_arithmetic.cpp.obj : warning LNK4006: "struct bf16x16 __cdecl load_kernel_t<struct bf16x16>(void const *)" (??$load_kernel_t@Ubf16x16@@@@YA?AUbf16x16@@PEBX@Z) already defined in vec_load.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_arithmetic.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct s8x16>(void *,struct s8x16)" (??$store_kernel_t@Us8x16@@@@YAXPEAXUs8x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_arithmetic.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct fp32x16>(void *,struct fp32x16)" (??$store_kernel_t@Ufp32x16@@@@YAXPEAXUfp32x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
  
C:\itrex\intel_extension_for_transformers\llm\runtime\graph\out\build\x64-release-avx2\vec_arithmetic.cpp.obj : warning LNK4006: "void __cdecl store_kernel_t<struct bf16x16>(void *,struct bf16x16)" (??$store_kernel_t@Ubf16x16@@@@YAXPEAXUbf16x16@@@Z) already defined in vec_store.cpp.obj; second definition ignored
```

Reply:
fix in [PR511](https://github.com/intel/intel-extension-for-transformers/pull/511)